PWGJE/TableProducer/emcalClusterHadronicCorrectionTask.cxx

```diff
@@ -0,0 +1,309 @@
+          auto trackPhiLow = -funcPtDepPhi.Eval(mom);
+
+          if ((dPhi < trackPhiHigh && dPhi > trackPhiLow) && fabs(dEta) < trackEtaMax) {
+            if (Nmatches == 0) {
+              closestTrkP = mom;
+            }
+            totalTrkP += mom;
+            Nmatches++;
+          }
+        } else {
+          // Do fixed dEta/dPhi matching (non-pT dependent)
+          if (fabs(dEta) >= minDEta || fabs(dPhi) >= minDPhi) {
+            continue; // Skip this track if outside the fixed cut region
+          }
+
+          // If track passes fixed dEta/dPhi cuts, process it
+          if (Nmatches == 0) {
+            closestTrkP = mom; // Closest track match
+          }
+          totalTrkP += mom; // Accumulate momentum
+          Nmatches++;       // Count this match
+        }
+
+      } // End of track loop
+      registry.fill(HIST("h2_ClsEvsNmatches"), cluster.energy(), Nmatches);
+
+      if (Nmatches >= 1) {
+        if (UseM02SubtractionScheme1) {
+          // Do M02-based correction if enabled
+          Ecluster1 = subtractM02ClusterEnergy(cluster.m02(), Ecluster1, Nmatches, closestTrkP, fHadCorr1, UseFraction1);
+          EclusterAll1 = subtractM02ClusterEnergy(cluster.m02(), EclusterAll1, Nmatches, totalTrkP, fHadCorralltrks1, UseFraction1);
+        } else {
+          // Default energy subtraction (100% and 70%)
+          Ecluster1 = subtractClusterEnergy(Ecluster1, closestTrkP, fHadCorr1, Nmatches, UseFraction1);
+          EclusterAll1 = subtractClusterEnergy(EclusterAll1, totalTrkP, fHadCorralltrks1, Nmatches, UseFraction1);
+        }
+
+        if (UseM02SubtractionScheme2) {
+          // Do M02-based correction if enabled
+          Ecluster2 = subtractM02ClusterEnergy(cluster.m02(), Ecluster2, Nmatches, closestTrkP, fHadCorr2, UseFraction2);
+          EclusterAll2 = subtractM02ClusterEnergy(cluster.m02(), EclusterAll2, Nmatches, totalTrkP, fHadCorralltrks2, UseFraction2);
+        } else {
+          // Default energy subtraction (100% and 70%)
+          Ecluster2 = subtractClusterEnergy(Ecluster2, closestTrkP, fHadCorr2, Nmatches, UseFraction2);
+          EclusterAll2 = subtractClusterEnergy(EclusterAll2, totalTrkP, fHadCorralltrks2, Nmatches, UseFraction2);
+        }
+      }
+      registry.fill(HIST("h_Ecluster1"), Ecluster1);
+      registry.fill(HIST("h_Ecluster2"), Ecluster2);
+      registry.fill(HIST("h_EclusterAll1"), EclusterAll1);
+      registry.fill(HIST("h_EclusterAll2"), EclusterAll2);
+      registry.fill(HIST("h2_ClsEvsEcluster1"), cluster.energy(), Ecluster1);
+      registry.fill(HIST("h2_ClsEvsEcluster2"), cluster.energy(), Ecluster2);
+      registry.fill(HIST("h2_ClsEvsEclusterAll1"), cluster.energy(), EclusterAll1);
+      registry.fill(HIST("h2_ClsEvsEclusterAll2"), cluster.energy(), EclusterAll2);
+
+      // Fill the table with all four corrected energies
+      hadroniccorrectedclusters(Ecluster1, Ecluster2, EclusterAll1, EclusterAll2);
```

Comment:
how does one match this table to the cluster table? What is the index? collision id?